pageserver/src/waldecoder.rs

```diff
@@ -86,10 +97,12 @@ impl WalStreamDecoder {
                 let hdr = XLogLongPageHeaderData::from_bytes(&mut self.inputbuf);
 
                 if hdr.std.xlp_pageaddr != self.lsn.0 {
-                    return Err(WalDecodeError {
-                        msg: "invalid xlog segment header".into(),
-                        lsn: self.lsn,
-                    });
+                    info!(
```

Comment:
Why is that okay now?

Reply:
Yet another result of pg_resetwal: we may receive not the segment we requested, but next segment.
All this stuff is annoying:(
But I failed to find some more elegant solution, especially taken in account that:
1. walproposer should be able to work both with main and nonrel_wal branches
2. compute node may work through wal_keeper or directly with pageserver.
3. It is not desirable to include third component (wal_keeper) in procedure of starting new node. And pageserver is not expected to hold all WAL.